README.md

```diff
@@ -1,7 +1,154 @@
 # Go Obscuro
-This is the reference implementation of the Obscuro Protocol.
 
-## Structure
+This repository contains the reference implementation of the [Obscuro Protocol](https://whitepaper.obscu.ro/).
+
+
+## High level overview
+
+The typical blockchain node runs multiple services in a single process. For example:
+- P2P Service
+- RPC Service
+- Data storage
+- Transaction execution
+- Mempool
+
+Obscuro uses TEEs (Intel SGX) to execute transactions in a confidential environment, which means we diverge from the typical architecture. 
+There are three main components of the architecture, each running as a separate process: the enclave, the host and the wallet extension.
+
+[//]: # (Todo - @joel - change the diagram to represent the wallet extension)
+![Architechture](design/obscuro_arch.jpeg)
+
+### I. The Enclave
+
+This is the core component of Obscuro which runs inside the TEE. 
+See [go/enclave](go/enclave)
+
+We use [EGo](https://www.edgeless.systems/products/ego/), an open source SDK for developing this confidential component.
+
+The Enclave exposes an [interface](go/common/enclave.go) over RPC which attempts to minimise the TCB (Trusted computing base).
+
+The enclave component has these main responsibilities:
+
+#### 1. Execute EVM transactions
+Obscuro has the goal to be fully compatible with the EVM, so smart contracts can be ported freely from other EVM compatible
+chains. To achieve this and minimise the effort and incompatibilities, we depend on [go-ethereum](https://github.com/ethereum/go-ethereum).
+
+The dependency on go-ethereum is not straight forward, since transaction execution is coupled with ethereum specific consensus rules,
+which had to be mocked out.
+
+See [go/enclave/evm](go/enclave/evm)
+
+
+#### 2. Store the state
+The dependency on go-ethereum for transaction execution, means that we have to use the same storage interfaces.
+
+In the current iteration we use [EdglessDB](https://www.edgeless.systems/products/edgelessdb/), an open source database tailor-made 
+for confidential computing.
+
+Go-ethereum uses a key-value store interface, which we implemented on top of the SQL database. The reason for this odd choice 
+is that the data needs to be indexed and encrypted befored being sent for storage. And the operation of storing data needs 
+to be resistant to side-channel analysis which would allow an attacker to infer information on what calculations are being made
+based on what data the enclave is requesting or storing.
+
+In a future iteration, we'll look at alternatives to connect to a performant k/v store designed or modified for confidential computing.
+
+See [go/enclave/db](go/enclave/db)
+
+
+#### 3. Consume Ethereum blocks 
+The enclave is fed ethereum blocks through the rpc interface. These blocks are used as the "Source of Truth", and the enclave 
+extracts useful information from them, such as published rollups, deposits to the bridge, etc. Ethereum Re-orgs have to be detected
```

Comment:
Re-orgs should not be capitalised. Even better, is there an alternative term which is more explanatory, like "Ethereum block reorganisation"

Reply:
re-org is a pretty widespread term. Whoever read this far, must be familiar with it